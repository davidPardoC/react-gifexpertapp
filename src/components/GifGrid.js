import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { loading, data:images } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__zoomIn'>{category}</h3>
      {loading ? (
        <p className='animate__animated animate__flash'>loading...</p>
      ) : (
        <div className="card-grid animate__fadeIn">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </div>
      )}
    </>
  );
};
