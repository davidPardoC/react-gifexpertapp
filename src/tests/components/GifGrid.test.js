import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import '@testing-library/jest-dom'
jest.mock("../../hooks/useFetchGifs");
describe("GifGrid component", () => {
  let category = "Goku";
  useFetchGifs.mockReturnValue({
    data: [],
    loading: true,
  });
  let wrapper = shallow(<GifGrid category={category} />);
  beforeEach(() => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    wrapper = shallow(<GifGrid category={category} />);
  });

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render items on data", () => {
    const gifs = [{ id: "ABC", url: "testUrl", title: "test" },{ id: "ABC123", url: "testUrl", title: "test" }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  });
});
