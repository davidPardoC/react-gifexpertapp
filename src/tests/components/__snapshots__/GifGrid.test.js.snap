// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component should match snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
</Fragment>
`;

exports[`GifGrid component should render items on data 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Goku
  </h3>
  <div
    className="card-grid animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test"
      url="testUrl"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="test"
      url="testUrl"
    />
  </div>
</Fragment>
`;
